### PR TITLE
Enable SIGNALFX_SERVER_TIMING_CONTEXT by default

### PR DESCRIPTION
**Pull request by pellared**
## Why

https://github.com/signalfx/gdi-specification/discussions/18#discussioncomment-554467

## What

- Enable `SIGNALFX_SERVER_TIMING_CONTEXT` by default.

### CODE REVIEW

**Comment by pellared** [comment]
All tests passed. Some CircleCI + GitHub integration issue: https://app.circleci.com/pipelines/github/pellared/signalfx-go-tracing/128/workflows/ebf77907-1e6f-4317-bfe7-5974d6411d71

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
+## [Unreleased] 1.8.0
+
+### Changed
+
+- Enable `SIGNALFX_SERVER_TIMING_CONTEXT` by default. ([#96](https://github.com/signalfx/signalfx-go-tracing/pull/96))
```
Comment (pellared) [inline CHANGELOG.md:13]: In theory, this is a breaking change but I think bumping to `v2` does not make sense in this case.
@MrAlias @owais  Do you agree?

Comment (owais) [reply CHANGELOG.md:13]: I agree. It's debatable whether this is a breaking change or not. As long as we don't overwrite any possible headers set by the user, I wouldn't call this a breaking change.

**Review by owais — APPROVED** [review]

File: contrib/internal/httputil/trace.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func TraceAndServe(h http.Handler, w http.ResponseWriter, r *http.Request, servi
 
 	w = wrapResponseWriter(w, span)
 
-	if strings.EqualFold("true", os.Getenv("SIGNALFX_SERVER_TIMING_CONTEXT")) {
+	if v := os.Getenv("SIGNALFX_SERVER_TIMING_CONTEXT"); v == "" || v == "1" || strings.EqualFold(v, "true") {
```
Comment (pellared) [inline contrib/internal/httputil/trace.go:46]: I think it should be:
```suggestion
if v := os.Getenv("SIGNALFX_SERVER_TIMING_CONTEXT"); v == "" || v != "0" || !strings.EqualFold(v, "false") {
```

do you agree @owais ?

Comment (owais) [reply contrib/internal/httputil/trace.go:46]: Do we have options or a config object in this repo? may move this condition there and have a check like `config.trace_response_header_enabled`.

**Review by owais — APPROVED** [review]

**Comment by owais** [comment]
LGTM @pellared

**Pull request merged**